pallets/identity/src/auth.rs

```diff
@@ -187,7 +196,24 @@ impl<T: Config> Module<T> {
         }
 
         // Run custom per-type validation and updates.
-        accepter(auth.authorization_data, auth.authorized_by)?;
+        let res = accepter(auth.authorization_data, auth.authorized_by);
+
+        if res.is_err() {
+            let mut updated_auth = Self::ensure_authorization(target, auth_id)?;
```

Comment:
The authorization value is already loaded in variable `auth`, which can be used instead.

Reply:
I originally used `auth` as it is but was getting an error that `value was partially moved` because of `let res = accepter(auth.authorization_data, auth.authorized_by);`